src/cpp/include/openvino/genai/llm_pipeline.hpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <optional>
+#include <variant>
+
+#include "openvino/core/any.hpp"
+#include "openvino/genai/generation_config.hpp"
+#include "openvino/genai/tokenizer.hpp"
+#include "openvino/genai/streamer_base.hpp"
+
+namespace ov {
+namespace genai {
+
+using StreamerVariant = std::variant<std::function<void(std::string)>, std::shared_ptr<StreamerBase>, std::monostate>;
+using OptionalGenerationConfig = std::optional<GenerationConfig>;
+using EncodedInputs = std::variant<ov::Tensor, std::pair<ov::Tensor, ov::Tensor>, TokenizedInputs>;
+using StringInputs = std::variant<std::string, std::vector<std::string>>;
+
+/**
+* @brief Structure to store resulting batched tokens and scores for each batch sequence
+*
+* @param tokens sequence of resulting tokens
+* @param scores scores for each sequence
+*/
+class EncodedResults {
+public:
+    std::vector<std::vector<int64_t>> tokens;
+    std::vector<float> scores;
+};
+
+/**
+* @brief Structure to store resulting batched text outputs and scores for each batch
+*
+* @param texts vector of resulting sequences
+* @param scores scores for each sequence
+*/
+class DecodedResults {
+public:
+    std::vector<std::string> texts;
+    std::vector<float> scores;
+
+     // @brief Convert DecodedResults to a vector of strings.
+     // @return A std::vector<std::string> containing the texts from the DecodedResults object.
+    operator std::string() const { 
+        OPENVINO_ASSERT(texts.size() == 1, "DecodedResults can be converted to string only if contains a single prompt");
+        return texts.at(0); 
+    }
+
+    // @brief Convert DecodedResults to a single string.
+    // @return std::string containing the texts from the DecodedResults object.
+    operator std::vector<std::string>() const { 
+        return texts; 
+    }
+    
+     // @brief Overloads operator<< to enhance output the contents of DecodedResults.
+     // @return A reference to the output stream with the concatenated texts.
+    friend std::ostream& operator<<(std::ostream& os, const DecodedResults& dr) {
+       for (size_t i = 0; i < dr.texts.size(); ++i) {
+            os << dr.texts[i];
+            if (i != dr.texts.size() - 1) {
+                os << std::endl;
+            }
+        }
+        return os;
+    }
+};
+
+/**
+* @brief This class is used for generation with LLMs.
+ */
+class OPENVINO_GENAI_EXPORTS LLMPipeline {
+public:
+    /**
+    * @brief Constructs an LLMPipeline from xml/bin files, tokenizers and configuration in the same dir.
+    *
+    * @param model_path Path to the dir model xml/bin files, tokenizers and generation_configs.json
+    * @param device optional device
+    * @param plugin_config optional plugin_config
+    */
+    LLMPipeline(
+        const std::string& path, 
```

Comment:
I guess we never though about it. C++ has `std::filesystem::path`. Why do we still use strings?